vcd/resource_vcd_nsxt_ip_set_test.go

```diff
@@ -0,0 +1,295 @@
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "ip_addresses.#", "5"),
+				),
+			},
+			resource.TestStep{
+				Config: configText1,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_ip_set.set1", "id", regexp.MustCompile(`^urn:vcloud:firewallGroup:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "name", "test-ip-set-changed"),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "description", ""),
+					resource.TestCheckTypeSetElemAttr("vcd_nsxt_ip_set.set1", "ip_addresses.*", "12.12.12.1"),
+					resource.TestCheckTypeSetElemAttr("vcd_nsxt_ip_set.set1", "ip_addresses.*", "2001:db6:0:0:0:0:0:0-2001:db6:0:ffff:ffff:ffff:ffff:ffff"),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "ip_addresses.#", "2"),
+				),
+			},
+			resource.TestStep{
+				Config: configText11,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_ip_set.set1", "id", regexp.MustCompile(`^urn:vcloud:firewallGroup:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "name", "test-ip-set-changed"),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "description", ""),
+					resource.TestCheckTypeSetElemAttr("vcd_nsxt_ip_set.set1", "ip_addresses.*", "12.12.12.1"),
+					resource.TestCheckTypeSetElemAttr("vcd_nsxt_ip_set.set1", "ip_addresses.*", "2001:db6:0:0:0:0:0:0-2001:db6:0:ffff:ffff:ffff:ffff:ffff"),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "ip_addresses.#", "2"),
+
+					resourceFieldsEqual("vcd_nsxt_ip_set.set1", "data.vcd_nsxt_ip_set.ds", []string{}),
+				),
+			},
+			// Test import with IP addresses
+			resource.TestStep{
+				ResourceName:      "vcd_nsxt_ip_set.set1",
+				ImportState:       true,
+				ImportStateVerify: true,
+				ImportStateIdFunc: importStateIdNsxtEdgeGatewayObject(testConfig, testConfig.Nsxt.EdgeGateway, "test-ip-set-changed"),
+			},
+			resource.TestStep{
+				Config: configText2,
+				Check: resource.ComposeAggregateTestCheckFunc(
+					resource.TestMatchResourceAttr("vcd_nsxt_ip_set.set1", "id", regexp.MustCompile(`^urn:vcloud:firewallGroup:.*$`)),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "name", "test-ip-set"),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "description", "test-ip-set-description"),
+					resource.TestCheckResourceAttr("vcd_nsxt_ip_set.set1", "ip_addresses.#", "0"),
+				),
+			},
+		},
+	})
+	postTestChecks(t)
+}
+
+const testAccNsxtIpSetPrereqs = `
+data "vcd_nsxt_edgegateway" "existing" {
+  org = "{{.Org}}"
+  vdc = "{{.NsxtVdc}}"
+
+  name = "{{.EdgeGw}}"
+}
+`
+
+const testAccNsxtIpSetEmpty = testAccNsxtIpSetPrereqs + `
+resource "vcd_nsxt_ip_set" "set1" {
+  org = "{{.Org}}"
+  vdc = "{{.NsxtVdc}}"
+
+  edge_gateway_id = data.vcd_nsxt_edgegateway.existing.id
```

Comment:
Total nit, but consider instead `existing` to use `existing_gw` just to depict the type when quickly scanning the code.

Reply:
Changed.